crates/ty_python_semantic/src/types/instance.rs

```diff
@@ -47,9 +52,11 @@ impl<'db> Type<'db> {
 /// A type representing the set of runtime objects which are instances of a certain nominal class.
 #[derive(Copy, Clone, Debug, Eq, PartialEq, Hash, salsa::Update)]
 pub struct NominalInstanceType<'db> {
+    pub(super) class: ClassType<'db>,
+
     // Keep this field private, so that the only way of constructing `NominalInstanceType` instances
     // is through the `Type::instance` constructor function.
-    class: ClassType<'db>,
+    _phantom: PhantomData<()>,
```

Comment:
oh, that's clever! It's a shame I'm only learning about this trick now -- much of the diff in #17525 came from me making the `class` field private, and I guess that was unnecessary